webui/react/src/pages/ExperimentDetails/ExperimentActions.tsx

```diff
@@ -29,91 +29,68 @@ interface Props {
-  const [ btnLoadingStates, setBtnLoadingStates ] = useState<ButtonLoadingStates>({
-    Activate: false,
-    Archive: false,
-    Cancel: false,
-    Fork: false,
-    Kill: false,
-    Pause: false,
-    Tensorboard: false,
-    Unarchive: false,
-  });
+  const [ isRunningActivate, setIsRunningActivate ] = useState<boolean>(false);
+  const [ isRunningArchive, setIsRunningArchive ] = useState<boolean>(false);
+  const [ isRunningCancel, setIsRunningCancel ] = useState<boolean>(false);
+  const [ isRunningKill, setIsRunningKill ] = useState<boolean>(false);
+  const [ isRunningPause, setIsRunningPause ] = useState<boolean>(false);
+  const [ isRunningUnarchive, setIsRunningUnarchive ] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsRunningArchive(false);
+    setIsRunningUnarchive(false);
+  }, [ experiment.archived ]);
+
+  useEffect(() => {
+    setIsRunningActivate(false);
+    setIsRunningCancel(false);
+    setIsRunningKill(false);
+    setIsRunningPause(false);
+  }, [ experiment.state ]);
 
   const handleArchive = useCallback(() => async (): Promise<void> => {
-    setBtnLoadingStates(state => ({ ...state, Archive: true }));
-    try {
-      await archiveExperiment({ experimentId: experiment.id });
-      onSettled();
-    } finally {
-      setBtnLoadingStates(state => ({ ...state, Archive: false }));
-    }
+    setIsRunningArchive(true);
+    await archiveExperiment({ experimentId: experiment.id });
+    onSettled();
   }, [ experiment.id, onSettled ]);
 
   const handleUnarchive = useCallback(() => async (): Promise<void> => {
-    setBtnLoadingStates(state => ({ ...state, Unarchive: true }));
-    try {
-      await unarchiveExperiment({ experimentId: experiment.id });
-      onSettled();
-    } finally {
-      setBtnLoadingStates(state => ({ ...state, Unarchive: false }));
-    }
+    setIsRunningUnarchive(true);
+    await unarchiveExperiment({ experimentId: experiment.id });
+    onSettled();
   }, [ experiment.id, onSettled ]);
 
   const handleKill = useCallback(async () => {
-    setBtnLoadingStates(state => ({ ...state, Kill: true }));
-    try {
-      await killExperiment({ experimentId: experiment.id });
-      onSettled();
-    } finally {
-      setBtnLoadingStates(state => ({ ...state, Kill: false }));
-    }
+    setIsRunningKill(true);
+    await killExperiment({ experimentId: experiment.id });
+    onSettled();
   }, [ experiment.id, onSettled ]);
 
   const handleCreateTensorboard = useCallback(async () => {
-    setBtnLoadingStates(state => ({ ...state, Tensorboard: true }));
-    try {
-      const tensorboard = await openOrCreateTensorboard({ experimentIds: [ experiment.id ] });
-      openCommand(tensorboard);
-      onSettled();
-    } finally {
-      setBtnLoadingStates(state => ({ ...state, Tensorboard: false }));
-    }
+    const tensorboard = await openOrCreateTensorboard({ experimentIds: [ experiment.id ] });
```

Comment:
With this:

> tensorboard loading state could on the request(s) that were made before it opened in a new tab (as it did before)

you mean that the code should be more similar to:
```javascript
const handleCreateTensorboard = useCallback(async () => {
  setIsRunningTensorboard(true);
  const tensorboard = await ...
  openCommand(tensorboard);
  onSettled();
  setIsRunningTensorboard(false);
}, [...]);
```

?

Reply:
that makes sense. it'd be easy to test it by introducing an artificial delay to requests using chrome dev tools.

Reply:
I've added this. Probably will need to be updated with `try ... catch ...` like the other ones, will continue the discussion on the other points.